components/dashboard/UpcomingInterviewDialog/UpcomingInterviewDialog.js

```diff
@@ -0,0 +1,250 @@
+      })
+      .catch(err => setSubmitError(err.message));
+  }
+
+  const {
+    handleSubmit,
+    handleChange,
+    handleChangeCustom,
+    values,
+    errors,
+    isSubmitting,
+    reset,
+  } = useFormValidation({}, upcomingInterviewFormValidation, submit);
+
+  const handleExited = () => {
+    reset();
+    setSubmitError();
+    setSuccess(false);
+  };
+
+  return (
+    <Dialog
+      fullWidth
+      open={show}
+      aria-labelledby="upcoming-interview-dialog"
+      onExited={handleExited}
+    >
+      <DialogTitle id="upcoming-interview-dialog" onClose={onClose}>
+        <Typography variant="h6">Have an upcoming interview?</Typography>
+        <Typography variant="body1" color="textSecondary">
+          If you have an interview coming up, let us know and we can send helpful guidance on how to
+          prepare.
+        </Typography>
+      </DialogTitle>
+      <DialogContent dividers>
+        <SubmitSuccess
+          message="Thank you for entering; we'll be providing some recommendations."
+          show={success}
+        />
+        {!(success || isSubmitting) && (
+          <form id={formId}>
+            <FormControl fullWidth error={!!errors.type}>
+              <InputLabel shrink>Interview Type</InputLabel>
+              <Select
+                displayEmpty
+                id={`${formId}-type`}
+                inputProps={{ name: 'type' }}
+                onChange={handleChange}
+                renderValue={type =>
+                  type ? (
+                    type.label
+                  ) : (
+                    <Typography variant="body1" className={classes.placeholder}>
+                      What kind of interview is it?
+                    </Typography>
+                  )
+                }
+                value={values.type || ''}
+              >
+                {INTERVIEW_TYPES.map(type => (
+                  <MenuItem value={type} key={type.value} className={classes.selectItem}>
+                    <Typography variant="body1">{type.label}</Typography>
+                    <Typography variant="caption">{type.description}</Typography>
+                  </MenuItem>
+                ))}
+              </Select>
+              {!!errors.type && <FormHelperText>{errors.type}</FormHelperText>}
+            </FormControl>
+            <FormControl>
+              <MuiPickersUtilsProvider utils={DateFnsUtils}>
+                <KeyboardDatePicker
+                  disablePast
+                  disableToolbar
+                  error={!!errors.date}
+                  format="MM/dd/yyyy"
+                  helperText={errors.date}
+                  id={`${formId}-date`}
+                  InputLabelProps={{ shrink: true }}
+                  KeyboardButtonProps={{ 'aria-label': 'Interview Date' }}
+                  label="Interview Date"
+                  margin="normal"
+                  onChange={d => handleChangeCustom('date', d)}
```

Comment:
Thoughts on doing something like [this](https://github.com/newjersey/career-network/blob/cf8a4e13b3221161957fa2744d5d3336d662316f/components/dashboard/ActivityInputDialog.js#L333) to keep "clean" dates in the DB (without meaningless random `HH:MM:SS` parts)?

Reply:
:+1: